src/header/common/link.rs

```diff
@@ -0,0 +1,622 @@
+    pub title_star: Option<String>,
+
+    /// Media Type: `type`
+    pub media_type: Option<Mime>,
+
+    /// Link Extension: `link-extensions`
+    pub link_extension: Option<String>
+}
+
+/// A Media Descriptors Enum based on
+/// https://www.w3.org/TR/html401/types.html#h-6.13
+#[derive(Clone, PartialEq, Debug)]
+pub enum MediaDesc {
+    /// screen
+    Screen,
+    /// tty
+    Tty,
+    /// tv
+    Tv,
+    /// projection
+    Projection,
+    /// handheld
+    Handheld,
+    /// print
+    Print,
+    /// braille
+    Braille,
+    /// aural
+    Aural,
+    /// all
+    All,
+    /// Other Values
+    Value(String)
+}
+
+/// A Link Relation Type Enum based on
+/// [RFC5988](https://tools.ietf.org/html/rfc5988#section-6.2.2)
+#[derive(Clone, PartialEq, Debug)]
+pub enum RelationType {
+    /// alternate
+    Alternate,
+    /// appendix
+    Appendix,
+    /// bookmark
+    Bookmark,
+    /// chapter
+    Chapter,
+    /// contents
+    Contents,
+    /// copyright
+    Copyright,
+    /// current
+    Current,
+    /// describedby
+    DescribedBy,
+    /// edit
+    Edit,
+    /// editMedia
+    EditMedia,
+    /// enclosure
+    Enclosure,
+    /// first
+    First,
+    /// glossary
+    Glossary,
+    /// help
+    Help,
+    /// hub
+    Hub,
+    /// index
+    Index,
+    /// last
+    Last,
+    /// latestVersion
+    LatestVersion,
+    /// license
+    License,
+    /// next
+    Next,
+    /// nextArchive
+    NextArchive,
+    /// payment
+    Payment,
+    /// prev
+    Prev,
+    /// predecessorVersion
+    PredecessorVersion,
+    /// previous
+    Previous,
+    /// prevArchive
+    PrevArchive,
+    /// related
+    Related,
+    /// replies
+    Replies,
+    /// section
+    Section,
+    /// self
+    RelationTypeSelf,
+    /// service
+    Service,
+    /// start
+    Start,
+    /// stylesheet
+    Stylesheet,
+    /// subsection
+    Subsection,
+    /// successorVersion
+    SuccessorVersion,
+    /// up
+    Up,
+    /// versionHistory
+    VersionHistory,
+    /// via
+    Via,
+    /// working-copy
+    WorkingCopy,
+    /// working-copy-of
+    WorkingCopyOf,
+    /// ext-rel-type
+    ExtRelType(Uri)
+}
+
+impl Link {
+    /// Create Link from URI-Reference
+    pub fn new(uri: &str) -> ::Result<Link> {
+        match Uri::new(uri) {
+            Err(_) => Err(::Error::Header),
+            Ok(u) => Ok(Link {
+                link: u,
+                rel: None,
+                anchor: None,
+                rev: None,
+                href_lang: Vec::new(),
+                media_desc: None,
+                title: None,
+                title_star: None,
+                media_type: None,
+                link_extension: None,
+            })
+        }
+    }
+}
+
+impl Header for Link {
+    fn header_name() -> &'static str {
+        static NAME: &'static str = "Link";
+        NAME
+    }
+
+    fn parse_header(raw: &Raw) -> ::Result<Link> {
+       from_one_raw_str(raw)
```

Comment:
Is there only allowed to be 1 `Link` header? If there a multiple, does the spec say anything about that? This function will parse if there is 1 link, and reject if there are more than 1, which is fine for some headers, just asking if that is the case with `Link`.